### ISSUE TITLE
OMV5 installation fails on a fresh Debian 10 Buster with btrfs root

### ISSUE THREAD

**Issue by danielepaolella** [openmediavault/openmediavault#414]
### Description of issue/question

Installation of OMV5 fails on a new install of Debian 10 Buster, having `btrfs` as root filesystem.

### Steps to reproduce issue

- Follow the instructions at https://forum.openmediavault.org/index.php/Thread/25062-Install-OMV5-on-Debian-10-Buster/ until:

      apt-get --yes --auto-remove --show-upgraded \
      	--allow-downgrades --allow-change-held-packages \
      	--no-install-recommends \
      	--option Dpkg::Options::="--force-confdef" \
      	--option DPkg::Options::="--force-confold" \
      	install openmediavault-keyring openmediavault

  fails with:

      Setting up system ...
      [WARNING ] /usr/lib/python3/dist-packages/salt/modules/file.py:32: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
        from collections import Iterable, Mapping
      
      [ERROR   ] Rendering exception occurred
      Traceback (most recent call last):
        File "/usr/lib/python3/dist-packages/salt/utils/templates.py", line 393, in render_jinja_tmpl
          output = template.render(**decoded_context)
        File "/usr/lib/python3/dist-packages/jinja2/asyncsupport.py", line 76, in render
          return original_render(self, *args, **kwargs)
        File "/usr/lib/python3/dist-packages/jinja2/environment.py", line 1008, in render
          return self.environment.handle_exception(exc_info, True)
        File "/usr/lib/python3/dist-packages/jinja2/environment.py", line 780, in handle_exception
          reraise(exc_type, exc_value, tb)
        File "/usr/lib/python3/dist-packages/jinja2/_compat.py", line 37, in reraise
          raise value.with_traceback(tb)
        File "<template>", line 57, in top-level template code
        File "/var/cache/salt/minion/extmods/modules/omv_utils.py", line 173, in get_root_filesystem
          fs = openmediavault.fs.Filesystem.from_mount_point('/')
        File "/usr/lib/python3/dist-packages/openmediavault/fs/__init__.py", line 95, in from_mount_point
          device = pyudev.Devices.from_device_number(context, 'block', st.st_dev)
        File "/usr/lib/python3/dist-packages/pyudev/device/_device.py", line 182, in from_device_number
          raise DeviceNotFoundByNumberError(typ, number)
      pyudev.device._errors.DeviceNotFoundByNumberError: No block device with number 22

### Additional debug info

- `cat /etc/fstab`

      # /etc/fstab: static file system information.
      #
      # Use 'blkid' to print the universally unique identifier for a
      # device; this may be used with UUID= as a more robust way to name devices
      # that works even if disks are added and removed. See fstab(5).
      #
      # <file system> <mount point>   <type>  <options>       <dump>  <pass>
      # / was on /dev/sda1 during installation
      UUID=c29477ed-9bb2-4056-b91a-c34580ec3759 /               btrfs   defaults        0       0
      # swap was on /dev/sda5 during installation
      UUID=62094831-436c-417c-8e00-a0733447bdb3 none            swap    sw              0       0

- `stat /`

      File: /
        Size: 276         Blocks: 32         IO Block: 4096   directory
      Device: 16h/22d Inode: 256         Links: 1
      Access: (0755/drwxr-xr-x)  Uid: (    0/    root)   Gid: (    0/    root)
      Access: 2019-08-07 11:09:01.836276415 +0200
      Modify: 2019-08-07 11:05:06.986117941 +0200
      Change: 2019-08-07 11:05:06.986117941 +0200
       Birth: -

- `python3`

      Python 3.7.3 (default, Apr  3 2019, 05:39:12) 
      [GCC 8.3.0] on linux
      Type "help", "copyright", "credits" or "license" for more information.
      >>> import os
      >>> st = os.stat("/")
      >>> st
      os.stat_result(st_mode=16877, st_ino=256, st_dev=22, st_nlink=1, st_uid=0, st_gid=0, st_size=276, st_atime=1565168941, st_mtime=1565168706, st_ctime=1565168706)
      >>> os.major(st.st_dev), os.minor(st.st_dev)
      (0, 22)

- `ls -alh /sys/dev/block/0:22`

      ls: cannot access '/sys/dev/block/0:22': No such file or directory

- `ls -alh /sys/dev/block`

      total 0
      drwxr-xr-x 2 root root 0 Aug  7 11:28 .
      drwxr-xr-x 4 root root 0 Aug  7 11:28 ..
      lrwxrwxrwx 1 root root 0 Aug  7 11:28 8:0 -> ../../devices/pci0000:00/0000:00:1f.2/ata1/host0/target0:0:0/0:0:0:0/block/sda
      lrwxrwxrwx 1 root root 0 Aug  7 11:28 8:1 -> ../../devices/pci0000:00/0000:00:1f.2/ata1/host0/target0:0:0/0:0:0:0/block/sda/sda1
      lrwxrwxrwx 1 root root 0 Aug  7 11:28 8:16 -> ../../devices/pci0000:00/0000:00:1f.2/ata1/host0/target0:0:1/0:0:1:0/block/sdb
      lrwxrwxrwx 1 root root 0 Aug  7 11:28 8:17 -> ../../devices/pci0000:00/0000:00:1f.2/ata1/host0/target0:0:1/0:0:1:0/block/sdb/sdb1
      lrwxrwxrwx 1 root root 0 Aug  7 11:28 8:2 -> ../../devices/pci0000:00/0000:00:1f.2/ata1/host0/target0:0:0/0:0:0:0/block/sda/sda2
      lrwxrwxrwx 1 root root 0 Aug  7 11:28 8:32 -> ../../devices/pci0000:00/0000:00:1f.2/ata2/host1/target1:0:0/1:0:0:0/block/sdc
      lrwxrwxrwx 1 root root 0 Aug  7 11:28 8:33 -> ../../devices/pci0000:00/0000:00:1f.2/ata2/host1/target1:0:0/1:0:0:0/block/sdc/sdc1
      lrwxrwxrwx 1 root root 0 Aug  7 11:28 8:5 -> ../../devices/pci0000:00/0000:00:1f.2/ata1/host0/target0:0:0/0:0:0:0/block/sda/sda5

**Comment by votdev** [comment]
I want to prohibit the use of ``findmnt``, so need to have more info if there is a way to get the necessary data. Can you please execute ``cat /proc/self/mountinfo`` and post the output.

**Comment by danielepaolella** [comment]
Here is `cat /proc/self/mountinfo`:

```19 24 0:18 / /sys rw,nosuid,nodev,noexec,relatime shared:7 - sysfs sysfs rw
20 24 0:4 / /proc rw,nosuid,nodev,noexec,relatime shared:14 - proc proc rw
21 24 0:6 / /dev rw,nosuid,relatime shared:2 - devtmpfs udev rw,size=2972632k,nr_inodes=743158,mode=755
22 21 0:19 / /dev/pts rw,nosuid,noexec,relatime shared:3 - devpts devpts rw,gid=5,mode=620,ptmxmode=000
23 24 0:20 / /run rw,nosuid,noexec,relatime shared:5 - tmpfs tmpfs rw,size=597508k,mode=755
24 0 0:21 / / rw,relatime shared:1 - btrfs /dev/sda1 rw,space_cache,subvolid=5,subvol=/
25 19 0:7 / /sys/kernel/security rw,nosuid,nodev,noexec,relatime shared:8 - securityfs securityfs rw
26 21 0:23 / /dev/shm rw,nosuid,nodev shared:4 - tmpfs tmpfs rw
27 23 0:24 / /run/lock rw,nosuid,nodev,noexec,relatime shared:6 - tmpfs tmpfs rw,size=5120k
28 19 0:25 / /sys/fs/cgroup ro,nosuid,nodev,noexec shared:9 - tmpfs tmpfs ro,mode=755
29 28 0:26 / /sys/fs/cgroup/unified rw,nosuid,nodev,noexec,relatime shared:10 - cgroup2 cgroup2 rw,nsdelegate
30 28 0:27 / /sys/fs/cgroup/systemd rw,nosuid,nodev,noexec,relatime shared:11 - cgroup cgroup rw,xattr,name=systemd
31 19 0:28 / /sys/fs/pstore rw,nosuid,nodev,noexec,relatime shared:12 - pstore pstore rw
32 19 0:29 / /sys/fs/bpf rw,nosuid,nodev,noexec,relatime shared:13 - bpf bpf rw,mode=700
33 28 0:30 / /sys/fs/cgroup/devices rw,nosuid,nodev,noexec,relatime shared:15 - cgroup cgroup rw,devices
34 28 0:31 / /sys/fs/cgroup/net_cls,net_prio rw,nosuid,nodev,noexec,relatime shared:16 - cgroup cgroup rw,net_cls,net_prio
35 28 0:32 / /sys/fs/cgroup/cpu,cpuacct rw,nosuid,nodev,noexec,relatime shared:17 - cgroup cgroup rw,cpu,cpuacct
36 28 0:33 / /sys/fs/cgroup/cpuset rw,nosuid,nodev,noexec,relatime shared:18 - cgroup cgroup rw,cpuset
37 28 0:34 / /sys/fs/cgroup/freezer rw,nosuid,nodev,noexec,relatime shared:19 - cgroup cgroup rw,freezer
38 28 0:35 / /sys/fs/cgroup/perf_event rw,nosuid,nodev,noexec,relatime shared:20 - cgroup cgroup rw,perf_event
39 28 0:36 / /sys/fs/cgroup/blkio rw,nosuid,nodev,noexec,relatime shared:21 - cgroup cgroup rw,blkio
40 28 0:37 / /sys/fs/cgroup/rdma rw,nosuid,nodev,noexec,relatime shared:22 - cgroup cgroup rw,rdma
41 28 0:38 / /sys/fs/cgroup/memory rw,nosuid,nodev,noexec,relatime shared:23 - cgroup cgroup rw,memory
42 28 0:39 / /sys/fs/cgroup/pids rw,nosuid,nodev,noexec,relatime shared:24 - cgroup cgroup rw,pids
43 20 0:40 / /proc/sys/fs/binfmt_misc rw,relatime shared:25 - autofs systemd-1 rw,fd=27,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=10313
44 21 0:17 / /dev/mqueue rw,relatime shared:26 - mqueue mqueue rw
45 21 0:41 / /dev/hugepages rw,relatime shared:27 - hugetlbfs hugetlbfs rw,pagesize=2M
46 19 0:8 / /sys/kernel/debug rw,relatime shared:28 - debugfs debugfs rw
187 19 0:47 / /sys/fs/fuse/connections rw,relatime shared:107 - fusectl fusectl rw
108 23 0:43 / /run/rpc_pipefs rw,relatime shared:94 - rpc_pipefs sunrpc rw
113 20 0:48 / /proc/fs/nfsd rw,relatime shared:97 - nfsd nfsd rw
118 43 0:49 / /proc/sys/fs/binfmt_misc rw,relatime shared:100 - binfmt_misc binfmt_misc rw
120 23 0:46 / /run/user/1000 rw,nosuid,nodev,relatime shared:104 - tmpfs tmpfs rw,size=597504k,mode=700,uid=1000,gid=1000
```

**Comment by danielepaolella** [comment]
Also, not my area of expertise, but this might be relevant:
https://btrfs.wiki.kernel.org/index.php/SysadminGuide#Description

**Comment by votdev** [comment]
> Also, not my area of expertise, but this might be relevant:
> [btrfs.wiki.kernel.org/index.php/SysadminGuide#Description](https://btrfs.wiki.kernel.org/index.php/SysadminGuide#Description)

That's the reason for the bug:
A Btrfs subvolume is an independently mountable POSIX filetree and not a block device (and cannot be treated as one).

**Comment by danielepaolella** [comment]
If you agree I'd update title and description of the report to better fit the topic.

**Comment by votdev** [comment]
> If you agree I'd update title and description of the report to better fit the topic.

Feel free to do that

**Comment by votdev** [comment]
Fixed in openmediavault 5.0.5.
